src/app/components/Drbb/DrbbContainer.jsx

```diff
@@ -5,42 +5,27 @@ import { Link } from 'react-router';
 
 import appConfig from '../../data/appConfig';
 import DrbbResult from './DrbbResult';
+import Store from '../../stores/Store';
+import { getResearchNowQueryString } from '../../utils/researchNowUtils';
 
 class DrbbContainer extends React.Component {
   constructor(props, context) {
     super();
     this.search = context.router.location.search || '';
+    const {
+      works,
+      totalWorks,
+      researchNowQueryString
+    } = Store.getState().drbbResults;
```

Comment:
Only `Application` listens to the Store. The other components can pull in data from the Store directly. For some reason, `Application` stores the Store in state as `data` and then passes it to the children. So, that is why the child components get there data from props instead of the Store. @EdwinGuzman could you explain this choice at all? I'm curious if their is a benefit.

Since `DrbbContainer` is further down the component hierarchy, I initially chose for this approach to just get from the Store directly.

Reply:
Yes, the top-down approach of fetching data and sending it to the DOM tree's children was the favorable approach a while back. Doing so, you can reduce renders if you check the props that come in from the parent if they haven't changed. Redux changed this when the approach became, "connect all the children components to the store directly and get data like that". That way, the prop-drilling goes away and components have fewer prop dependencies that they might not even use (i.e. passing down a prop through a few components just because _one_ component uses).
The approach you're doing seems fine. You can also try using the new context API at the root, and then "connecting" the child component to it so it can get the data.